internal/test_cases/run_test_case.go

```diff
@@ -17,8 +18,45 @@ import (
 // It is sent to the "run" command of the executable,
 // the expected outputs are generated using the lox.Run function,
 // With that the output of the executable is matched.
+
 type RunTestCase struct {
 	FileContents string
+	FrontMatter  RunTestCaseFrontMater
+}
+
+type RunTestCaseFrontMater struct {
+	ExpectedErrorType string `yaml:"error_type"`
+}
+
+func NewRunTestCaseFromFileContents(fileContents []byte, filePath string) RunTestCase {
+	contents := string(fileContents)
+
+	if contents[:4] != "---\n" {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: %s has malformed frontmatter: no beginning triple dashes", filePath))
+	}
+
+	contents = contents[4:]
+	endingTripleDashIndex := strings.Index(contents, "\n---\n")
+	if endingTripleDashIndex == -1 {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: %s has malformed frontmatter: no ending triple dashes", filePath))
+	}
+
+	frontMatterBytes := []byte(contents[:endingTripleDashIndex])
+	contents = contents[endingTripleDashIndex+5:]
+
+	var frontMatter RunTestCaseFrontMater
+	err := yaml.Unmarshal(frontMatterBytes, &frontMatter)
+	if err != nil {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: %s has malformed frontmatter: can't unmarshal", filePath))
+	}
+	if frontMatter.ExpectedErrorType == "" {
```

Comment:
To prevent mistakes, let's check if the value is valid here too - assert that it matches the list of expected values we support.